src/overlays/actors/ovl_Demo_Sa/z_demo_sa.c

```diff
@@ -21,155 +78,762 @@ const ActorInit Demo_Sa_InitVars = {
+}
+
+s32 func_8098E6A0(DemoSa* this, GlobalContext* globalCtx, u16 arg2, s32 arg3) {
+    CsCmdActorAction* npcAction = DemoSa_GetNpcAction(globalCtx, arg3);
+
+    if ((npcAction != NULL) && (npcAction->action != arg2)) {
+        return 1;
+    }
+    return 0;
+}
+
+void func_8098E6EC(DemoSa* this, GlobalContext* globalCtx, s32 actionIdx) {
+    CsCmdActorAction* npcAction = DemoSa_GetNpcAction(globalCtx, actionIdx);
+
+    if (npcAction != NULL) {
+        this->actor.posRot.pos.x = npcAction->startPos.x;
+        this->actor.posRot.pos.y = npcAction->startPos.y;
+        this->actor.posRot.pos.z = npcAction->startPos.z;
+        this->actor.posRot.rot.y = this->actor.shape.rot.y = npcAction->rot.y;
+    }
+}
+
+void func_8098E76C(DemoSa* this, AnimationHeader* animationHeader, u8 arg2, f32 transitionRate, s32 arg4) {
+    s32 pad[2];
+    s16 frameCount = SkelAnime_GetFrameCount(&animationHeader->genericHeader);
+    f32 playbackSpeed;
+    f32 unk0;
+    f32 fc;
+
+    if (arg4 == 0) {
+        unk0 = 0.0f;
+        fc = frameCount;
+        playbackSpeed = 1.0f;
+    } else {
+        fc = 0.0f;
+        unk0 = frameCount;
+        playbackSpeed = -1.0f;
+    }
+
+    SkelAnime_ChangeAnim(&this->skelAnime, animationHeader, playbackSpeed, unk0, fc, arg2, transitionRate);
+}
+
+void func_8098E7FC(DemoSa* this, GlobalContext* globalCtx) {
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600B1A0, &D_060021D8, NULL, NULL, 0);
+    this->actor.shape.unk_08 = -10000.0f;
+    func_8098E508(this, 1);
+    func_8098E51C(this, 0);
+}
+
+void func_8098E86C(DemoSa* this, GlobalContext* globalCtx) {
+    Vec3f* posRot = &this->actor.posRot.pos;
+    f32 posX = posRot->x;
+    f32 posY = posRot->y;
+    f32 posZ = posRot->z;
+
+    Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, posX, posY, posZ, 0, 0, 0, 2);
+}
+
+void func_8098E8C8(DemoSa* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 posX = player->actor.posRot.pos.x;
+    f32 posY = player->actor.posRot.pos.y + 80.0f;
+    f32 posZ = player->actor.posRot.pos.z;
+
+    Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_EFFECT, posX, posY, posZ, 0, 0, 0,
+                        0xB);
+    Item_Give(globalCtx, ITEM_MEDALLION_FOREST);
+}
+
+void func_8098E944(DemoSa* this, GlobalContext* globalCtx) {
+    this->actor.shape.unk_08 += (250.0f / 3.0f);
+}
+
+void func_8098E960(DemoSa* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Player* player;
+    s16 newRotY;
+
+    if ((gSaveContext.chamberCutsceneNum == 0) && (gSaveContext.sceneSetupIndex < 4)) {
+        player = PLAYER;
+        this->action = 1;
+        globalCtx->csCtx.segment = D_8099010C;
+        gSaveContext.cutsceneTrigger = 2;
+        Item_Give(globalCtx, ITEM_MEDALLION_FOREST);
+
+        newRotY = this->actor.posRot.rot.y + 0x8000;
+        player->actor.shape.rot.y = newRotY;
+        player->actor.posRot.rot.y = newRotY;
```

Comment:
Similar to something I suggested in En_Zl2 but any chance this matches?
```suggestion
        player->actor.posRot.rot.y = player->actor.shape.rot.y = this->actor.posRot.rot.y + 0x8000;
```